src/main/java/it/gov/pagopa/payment/service/payment/common/CommonCancelServiceImpl.java

```diff
@@ -51,17 +52,17 @@ public void cancelTransaction(String trxId, String merchantId, String acquirerId
             TransactionInProgress trx = repository.findByIdThrottled(trxId);
 
             if (trx == null) {
-                throw new ClientExceptionNoBody(HttpStatus.NOT_FOUND, "[CANCEL_TRANSACTION] Cannot found transaction having id: " + trxId);
+                throw new TransactionNotFoundOrExpiredException(ExceptionCode.TRX_NOT_FOUND_OR_EXPIRED, "[CANCEL_TRANSACTION] Cannot found transaction having id: " + trxId);
```

Comment:
some Exception, like this, could have a default code that is supposed to be always the same: we could thus expose a constructor without CODE, using such default code in the constructor